File: 2657. Find the Prefix Common Array of Two Arrays/2657.myans.go
func findThePrefixCommonArray(A []int, B []int) []int {
    n := len(A)
    exists := make(map[int]bool)
    common := 0
    answer := make([]int, n)

    for i := range A {
        x := A[i]
        y := B[i]

        if exists[x] {
            common += 1
        }
        exists[x] = true

        if exists[y] {
            common += 1
        }
        exists[y] = true

        answer[i] = common
    }

    return answer
}

File: 2657. Find the Prefix Common Array of Two Arrays/2657.myans.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        std::set<int> exists;
        std::vector<int> prefixCommonArray(A.size());
        int common = 0;

        for (int i = 0; i < A.size(); i++) {
            if (!exists.insert(A[i]).second) {
                common += 1;
            }

            if (!exists.insert(B[i]).second) {
                common += 1;
            }

            prefixCommonArray[i] = common;
        }

        return prefixCommonArray;
    }
};